src/crons/retry-error.rb

```diff
@@ -0,0 +1,89 @@
+#!/bin/ruby
+require 'yaml'
+require 'fileutils'
+require 'net/smtp'
+
+#  retry-error
+# ==================================================
+# Name : David J. Davis
+# Date :  4/11/2017
+#
+# Requirements:  You must run add_project_env.sh first
+# so that the environmental variable is able to be used.
+#
+# Useage: From the terminal as root user run the command
+# ruby ./retry-error.rb {filename.yaml}
+#
+# Description: The following code goes through the following
+# checks to make sure that all items are present
+# ---- ENV['HYDRA_PROJECT_NAME']
+# ---- ARGV of a filename which is the error file
+# ---- Error File Exists
+# ---- Project Names Match
+#  If all exists it tries to run through the export script
+#  and if it passes the file is moved out of the error folder
+#  to the completed folder, if fails it is not moved.
+
+# Make Sure that the ENV variables are set
+if (!ENV.include? 'HYDRA_PROJECT_NAME') then
+  abort "Missing ENV variable, 'HYDRA_PROJECT_NAME'"
+end
+
+# We are expecting to run a single file again by hand.
+# Expecting this check to make sure that there is an argument given
+if ARGV.empty?
+  puts "Error: Expected filename"
+  abort "Usage: ruby ./retry-error.rb {filename.yaml}"
+end
+
+# Don't let them add more arguments than needed
+if ARGV.length >= 2
+  puts "Error: Too many arguments supplied"
+  abort "Usage: ruby ./retry-error.rb {filename.yaml}"
+end
+
+# define the filename from the arguments
+filename = ARGV
```

Comment:
you didn't change it. :)

Perl does the same thing. 
`my($first, $second, $third) = @my_array;`

In some situations it makes sense. In some it doesn't.  I would argue that in this case it doesnt make sense. If your style is going to include the verbose commenting, it makes more sense to break it up. It makes your commenting easier and more intuitive. 

Also, it really isn't needed. You could just be using ARGV[0] instead of assigning the file name, since you already have the filename in a variable. 

essentially what you are dong is:
```
a = 1
b = a
puts b
```